internal/integrationtest/compile_1/compile_test.go

```diff
@@ -112,6 +115,35 @@ func compileErrorMessage(t *testing.T, env *integrationtest.Environment, cli *in
 }
 
 func compileWithSimpleSketch(t *testing.T, env *integrationtest.Environment, cli *integrationtest.ArduinoCLI) {
+	compileWithSimpleSketchCustomEnv(t, env, cli, cli.GetDefaultEnv())
+}
+
+func compileWithCachePurgeNeeded(t *testing.T, env *integrationtest.Environment, cli *integrationtest.ArduinoCLI) {
+	// create directories that must be purged
+	baseDir := paths.TempDir().Join("arduino", "sketches")
+
+	// purge case: hold file too old
+	oldDir1 := baseDir.Join("test_old_sketch_1")
+	require.NoError(t, os.MkdirAll(oldDir1.String(), 0770))
+	require.NoError(t, oldDir1.Join(".last-used").WriteFile([]byte{}))
+	require.NoError(t, os.Chtimes(oldDir1.Join(".last-used").String(), time.Now(), time.Unix(0, 0)))
+	// purge case: hold file not existing
+	oldDir2 := baseDir.Join("test_old_sketch_2")
+	require.NoError(t, os.MkdirAll(oldDir2.String(), 0770))
+
+	defer os.RemoveAll(oldDir1.String())
+	defer os.RemoveAll(oldDir2.String())
+
+	customEnv := cli.GetDefaultEnv()
+	customEnv["ARDUINO_BUILD_CACHE_COMPILATIONS_BEFORE_PURGE"] = "1"
+	compileWithSimpleSketchCustomEnv(t, env, cli, customEnv)
+
+	// check that purge has been run
+	require.NoFileExists(t, oldDir1.String())
+	require.NoFileExists(t, oldDir2.String())
```

Comment:
If the `.last-used` file is missing, I prefer to not remove the directory.
I'm a bit conservative here, but I prefer to not delete the directory unless everything checks out.

```suggestion
	require.FileExists(t, oldDir2.String())
```